src/Sound/Tidal/Control.hs

```diff
@@ -412,38 +412,68 @@ sec p = (realToFrac <$> cF 1 "_cps") *| p
-ctrigger = triggerWith $ (fromIntegral :: Int -> Rational) . ceiling
-
-qtrigger :: Show a => a -> Pattern b -> Pattern b
+-- | (Alias @__qt__@) Quantise trigger. Aligns the start of the pattern
+-- with the next cycle boundary. For example, this pattern will fade in
+-- starting with the next cycle after the pattern is evaluated:
+-- 
+-- @
+-- d1 $ qtrigger $ s "hh(5, 8)" # amp envL
+-- @
+-- 
+-- Note that the pattern will start playing immediately. The /start/ of the
+-- pattern aligns with the next cycle boundary, but events will play before
+-- if the pattern has events at negative timestamps (which most loops do).
+-- These events can be filtered out, for example:
+-- 
+-- @
+-- d1 $ qtrigger $ filterWhen (>= 0) $ s "hh(5, 8)"
+-- @
+qtrigger :: Pattern a -> Pattern a
 qtrigger = ctrigger
 
-rtrigger :: Show a => a -> Pattern b -> Pattern b
+qt :: Pattern a -> Pattern a
+qt = qtrigger
+
+-- | Ceiling trigger. Aligns the start of a pattern to the next cycle
+-- boundary, just like 'qtrigger'.
+ctrigger :: Pattern a -> Pattern a
+ctrigger = triggerWith $ (fromIntegral :: Int -> Rational) . ceiling
+
+-- | Rounded trigger. Aligns the start of a pattern to the nearest cycle
+-- boundary, either next or previous.
+rtrigger :: Pattern a -> Pattern a
 rtrigger = triggerWith $ (fromIntegral :: Int -> Rational) . round
 
-ftrigger :: Show a => a -> Pattern b -> Pattern b
+-- | Floor trigger. Aligns the start of a pattern to the previous cycle
+-- boundary.
+ftrigger :: Pattern a -> Pattern a
 ftrigger = triggerWith $ (fromIntegral :: Int -> Rational) . floor
 
-qt :: Show a => a -> Pattern b -> Pattern b
-qt = qtrigger
-
-reset :: Show a => a -> Pattern b -> Pattern b
-reset k pat = pat {query = q}
-  where q st = query (rotR (offset st) $ when (<=0) (const silence) pat) st
-        f = (fromIntegral :: Int -> Rational) . floor
-        offset st = fromMaybe 0 $ do p <- Map.lookup ctrl (controls st)
-                                     return (f $ fromMaybe 0 $ getR p)
-        ctrl = "_t_" ++ show k
+-- | (Alias @__mt__@) Mod trigger. Aligns the start of a pattern to the
+-- next cycle boundary where the cycle is evenly divisible by a given
+-- number. 'qtrigger' is equivalent to @mtrigger 1@.
+mtrigger :: Int -> Pattern a -> Pattern a
+mtrigger n = triggerWith $ fromIntegral . nextMod
+  where nextMod t = n * ceiling (t / (fromIntegral n))
+
+mt :: Int -> Pattern a -> Pattern a
+mt = mtrigger
+
+-- | This aligns the start of a pattern to some value relative to the
+-- time the pattern is evaluated. The provided function maps the evaluation
+-- time (on the global cycle clock) to a new time, and then @triggerWith@
+-- aligns the pattern's start to the time that's returned.
+triggerWith :: (Time -> Time) -> Pattern a -> Pattern a
+triggerWith f pat = pat {query = q}
+  where q st = query (rotR (offset st) pat) st
+        offset st = fromMaybe 0 $ f <$> (Map.lookup ctrl (controls st) >>= getR)
+        ctrl = "_t_pattern"
```

Comment:
I would prefer that this string `"_t_pattern"` was defined as a constant.

Reply:
Sounds good, I'll do that.

Reply:
Okay, I pushed a final update that defines this ID string as a constant in `Stream.hs`, so that if it gets changed, `trigger` will continue to work.